### DISCUSSION TITLE
zh/docs/basic/try-with-docker/

### DISCUSSION THREAD

**Discussion by giscus** [casdoor/casdoor#3236] [Docs comments]
# zh/docs/basic/try-with-docker/

使用 Docker 运行

https://casdoor.org/zh/docs/basic/try-with-docker/

**Comment by wheat0r** [comment]
标准镜像配置sqlite时，db文件是否需要手动创建？
群晖DSM通过comopse部署时报错panic: unable to open database file: out of memory，已经按照 #3227 进行修改仍然存在相同错误。

**Reply by hsluoyz** [reply]
@wheat0r see: https://github.com/casdoor/casdoor/issues/3227#issuecomment-2367746599, you need to make sure your linux user has permission to create the sqlite db file in that path in the container. It's your responsibility to do it

**Reply by wheat0r** [reply]
> @wheat0r see: [#3227 (comment)](https://github.com/casdoor/casdoor/issues/3227#issuecomment-2367746599), you need to make sure your linux user has permission to create the sqlite db file in that path in the container. It's your responsibility to do it

实际上问题出在app.conf中，dataSourceName的正确写法是 `dataSourceName = "/conf/casdoor.db?cache=shared"` 而不是sqlite的URI格式file:/conf/casdoor.db?cache=shared。
当我尝试在容器中以/conf为工作目录运行/server的时候，我发现/conf目录中产生一个 `"'file:casdoor.db"` 文件，说明dataSourceName把file:当成路径的一部分了。

**Reply by dacongda** [reply]
> > @wheat0r see: [#3227 (comment)](https://github.com/casdoor/casdoor/issues/3227#issuecomment-2367746599), you need to make sure your linux user has permission to create the sqlite db file in that path in the container. It's your responsibility to do it
> 
> 实际上问题出在app.conf中，dataSourceName的正确写法是 `dataSourceName = "/conf/casdoor.db?cache=shared"` 而不是sqlite的URI格式file:/conf/casdoor.db?cache=shared。 当我尝试在容器中以/conf为工作目录运行/server的时候，我发现/conf目录中产生一个 `"'file:casdoor.db"` 文件，说明dataSourceName把file:当成路径的一部分了。

may be you have wrote a wrong conf, I use uri with `file:` prefix and successful create a db in conf, I think you used `dataSourceName = "'file:/conf/casdoor.db?cache=shared"` 
![image](https://github.com/user-attachments/assets/7481c99e-665c-4e9e-ade6-bb9c72a2b939)
![image](https://github.com/user-attachments/assets/aec1b1d1-28ee-4e6b-ac15-5411f30e8574)

**Reply by dacongda** [reply]
> > @wheat0r see: [#3227 (comment)](https://github.com/casdoor/casdoor/issues/3227#issuecomment-2367746599), you need to make sure your linux user has permission to create the sqlite db file in that path in the container. It's your responsibility to do it
> 
> 实际上问题出在app.conf中，dataSourceName的正确写法是 `dataSourceName = "/conf/casdoor.db?cache=shared"` 而不是sqlite的URI格式file:/conf/casdoor.db?cache=shared。 当我尝试在容器中以/conf为工作目录运行/server的时候，我发现/conf目录中产生一个 `"'file:casdoor.db"` 文件，说明dataSourceName把file:当成路径的一部分了。

btw, DSN with `file:` prefix is used by pacakge `modernc.org/sqlite`, in [modernc.org/sqlite](https://gitlab.com/cznic/sqlite/-/blob/master/sqlite.go?ref_type=heads) you can see it proccess file: prefix in line 821

![image](https://github.com/user-attachments/assets/5ccf0552-98a6-406e-ba78-53083de54a7e)

**Comment by theSemenov** [comment]
> docker run  -p 8000:8000 -v /folder/of/app.conf:/conf casbin/casdoor:latest
> Create conf/app.conf. You can copy it from conf/app.conf in Casdoor. For more details about app.conf, you can see Via Ini file.

Let's run docker with exposed port 8080 with downloaded conf/app.conf.
Than go to discovery page http://localhost:8000/.well-known/openid-configuration
It's seems that port conflict with something internal. In discovery I see wrong port for authorization_endpoint (7001 instead of 8000)

> "authorization_endpoint": "http://localhost:7001/login/oauth/authorize",
> "token_endpoint": "http://localhost:8000/api/login/oauth/access_token",

This fixes when change port in app.conf (httpport = 8001) and expose port from docker
> docker run  -p 8001:8001 -v /folder/of/app.conf:/conf casbin/casdoor:latest

It is a workaround, but maybe it should be described in the documentation and examples for everything to work out of the box

**Reply by dacongda** [reply]
to fix this easily, you can simply change `originFrontend` and `origin` in app.conf

for example, your frontend domain is `example.com` and using port 8001, you should set originFrontend to `http://example.com:8001`
your backend domain is `example.com` and using port 8002, you should set origin to `http://example.com:8002`
